src/bank.rs

```diff
@@ -942,23 +969,79 @@ impl Bank {
     pub fn set_finality(&self, finality: usize) {
         self.finality_time.store(finality, Ordering::Relaxed);
     }
+
+    pub fn store_account_subscription(
+        &self,
+        bank_sub_id: Pubkey,
+        pubkey: Pubkey,
+        sink: Sink<Account>,
+    ) {
+        let mut subscriptions = self.account_subscriptions.write().unwrap();
+        subscriptions.insert(bank_sub_id, (pubkey, sink));
+    }
+
+    pub fn remove_account_subscription(&self, bank_sub_id: &Pubkey) -> bool {
+        let mut subscriptions = self.account_subscriptions.write().unwrap();
+        subscriptions.remove(bank_sub_id).is_some()
+    }
+
+    fn check_account_subscriptions(&self, pubkey: &Pubkey, pre_userdata: &[u8], account: &Account) {
+        for (_bank_sub_id, (id, sink)) in self.account_subscriptions.read().unwrap().iter() {
+            if pubkey == id && pre_userdata != account.userdata.as_slice() {
+                sink.notify(Ok(account.clone())).wait().unwrap();
+            }
+        }
+    }
+
+    pub fn store_signature_subscription(
```

Comment:
The name `store_signature_subscription` confused me for a moment.  Perhaps `add_signature_subscription` (same comment for `store_account_subscription`)